app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -67,6 +69,7 @@
   private final Queue<AnalyticJob> _retryQueue = new ConcurrentLinkedQueue<AnalyticJob>();
 
   public void updateResourceManagerAddresses() {
+    String resourceManagerHttpPolicy = configuration.get(RESOURCE_MANAGER_HTTP_POLICY);
```

Comment:
Can you set default to HTTP_ONLY, just in case resourceManagerHttpPolicy is null.

```String resourceManagerHttpPolicy = configuration.get(RESOURCE_MANAGER_HTTP_POLICY, HTTP_ONLY);```